plugins/leemons-plugin-leebrary/frontend/src/components/AssetList.js

```diff
@@ -71,283 +52,227 @@ function getOwner(asset) {
-      };
+      .map((item) => ({
+        value: item.key,
+        label: item.name,
+        icon: (
+          <Box style={{ height: 16, marginBottom: 5 }}>
+            <ImageLoader src={item.icon} style={{ width: 16, height: 16, position: 'relative' }} />
+          </Box>
+        ),
+      }));
+    return [{ label: t('labels.resourceTypes'), value: 'all' }, ...filteredCategories];
+  }, [allowCategoryFilter, categories]);
 
-      if (categoryProp?.key?.includes('leebrary-subject')) {
-        delete query.category;
-      }
-      if (categoryProp?.key === 'leebrary-shared') {
-        delete query.category;
-        query.onlyShared = true;
-      }
+  const getMediaTypeSelectData = useMemo(() => {
+    if (!mediaTypes?.length) return null;
+    return [{ label: t('labels.resourceTypes'), value: 'all' }, ...mediaTypes];
+  }, [mediaTypes]);
 
-      if (categoryProp?.key === 'leebrary-recent') {
-        query.roles = JSON.stringify(['owner']);
-      }
+  const getEmptyState = () => {
+    if (searchCriteriaDebounced && !isEmpty(searchCriteriaDebounced)) {
+      return <SearchEmpty t={t} />;
+    }
 
-      // TODO: Category filter should apply to leebrary-shared section as well
-      if (
-        multiCategorySections.includes(categoryProp?.key) &&
-        categoryProp?.key !== 'leebrary-shared'
-      ) {
-        delete query.category;
-        const activateCategoryFilter = categoryFilter && categoryFilter !== 'all';
-        if (activateCategoryFilter) {
-          const chosenCategory = find(store.categories, { key: categoryFilter });
-          if (chosenCategory) query.categoryFilter = chosenCategory.id;
-        }
-      }
+    return <ListEmpty t={t} isRecentPage={category?.key === 'leebrary-recent'} />;
+  };
 
-      const response = await getAssetsRequest(query);
+  // -------------------------------------------------------------------------------------
+  // DRAWER HANDLERS & TOOLBAR
 
-      const results = response?.assets || [];
-      setStoreValue('assets', uniqBy(results, 'asset'));
-      if (isEmpty(results)) {
-        setStoreValue('pageAssetsData', []);
-      }
-      clearAssetLoading();
-    } catch (err) {
-      clearAssetLoading();
-      addErrorAlert(getErrorMessage(err));
+  const showDrawer = useMemo(() => {
+    if (!assetListIsLoading && !isNil(selectedAsset) && !isEmpty(selectedAsset)) {
+      setIsDrawerOpen(true);
+    } else {
+      setIsDrawerOpen(false);
     }
-  }
-
-  async function loadAsset(id, forceLoad) {
-    try {
-      const item = find(store.pageAssetsData.items, { id });
 
-      if (item && !forceLoad) {
-        setStoreValue('asset', prepareAsset(item, published));
-      } else {
-        const response = await getAssetsByIdsRequest([id]);
-        if (!isEmpty(response?.assets)) {
-          const value = response.assets[0];
+    return !assetListIsLoading && !isNil(selectedAsset) && !isEmpty(selectedAsset);
+  }, [assetListIsLoading, selectedAsset, isDetailOpen]);
```

Comment:
Is showDrawer doing the same as isDrawerOpen?

Reply:
TBD